### PR TITLE
Add terrain provider ability

### PR DESCRIPTION
**Pull request by Sagie501**
Add map terrain provider component to add the ability of showing terrain on the cesium map

### CODE REVIEW

File: projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    switch (this.provider) {
+      case MapTerrainProviderOptions.CesiumTerrain:
+      case MapTerrainProviderOptions.ArcGISTiledElevation:
+      case MapTerrainProviderOptions.GoogleEarthEnterprise:
+      case MapTerrainProviderOptions.VRTheWorld:
+      case MapTerrainProviderOptions.Ellipsoid:
+        this.terrainProvider = new this.provider(this.options);
+        break;
+      case MapTerrainProviderOptions.WorldTerrain:
+        this.terrainProvider = this.provider(this.options);
```
Comment (eitanfr) [inline projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts]: Looks like world terrain and all other terrains have the same logic, so the switch case is redundant. Maybe you can change it to a simple if else

Comment (Sagie501) [reply projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts]: The world terrain provider function already returns a new instance. The other terrain providers is like class that should have new.

File: projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+   * refer to cesium docs for details https://cesiumjs.org/Cesium/Build/Documentation/TerrainProvider.html
+   */
+  @Input()
+  options: { url?: string } = {};
+
+  /**
+   * the provider
+   */
+  @Input()
+  provider: any;
```
[the remark this answers is not in the document]
Comment (Sagie501) [reply projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts]: I tried but the compiler says that the provider in not a Function type. Actually it's like in the ac-map-layer-provider component.

**Comment by Sagie501** [comment]
> Thx for the PR.
> Can you please add your changes to CHANGES.md file?
> Also looks like the CI fails because of lint, can you please fix it and make sure that the build process works.

I coudn't find CHANGES.md file. Do you mean CHANGELOG.md file?

**Comment by eitanfr** [comment]
yes CHANGELOG.md

File: projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts (hunk 1 above)
Comment (eitanfr) [reply projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts]: Indeed, I meant to use the `if` something like:
```
if (this.provider === MapTerrainProviderOptions.WorldTerrain) {
this.terrainProvider = this.provider(this.options);
} else {
// all others 
this.terrainProvider = new this.provider(this.options);
}
```

Comment (Sagie501) [reply projects/angular-cesium/src/lib/angular-cesium/components/ac-map-terrain-provider/ac-map-terrian-provider.component.ts]: But the switch case has default that has different case from the others, so If I will change it to if else I still need to check every provider option. In addition It maintains uniformity with the ac-map-layers-provider component that has the same behavior. If you are anyway prefer it in if else comment again and I will change it.

**Pull request merged**